test/api/test_workflows.py

```diff
@@ -3047,6 +3047,37 @@ def test_run_replace_params_by_tool(self):
         self.__assert_lines_hid_line_count_is(history_id, 2, 5)
         self.__assert_lines_hid_line_count_is(history_id, 3, 5)
 
+    @skip_without_tool("random_lines1")
+    def test_integer_param_replacement(self):
+        workflow_id = self._upload_yaml_workflow("""
+class: GalaxyWorkflow
+inputs:
+  test_input: data
+steps:
+  first:
+    tool_id: random_lines1
+    state:
+      num_lines: ${numlines}
+      input:
+        $link: test_input
+""", round_trip_format_conversion=True)
+        with self.dataset_populator.test_history() as history_id:
+            hda = self.dataset_populator.new_dataset(history_id, content="\n".join(str(_) for _ in range(10)))
```

Comment:
This needs to be a raw string, and then it actually works

Reply:
Really?  I snagged this straight from another test.

Reply:
Argh, I thought so, but for some reason num_lines falls back to 1 in this test, making me think that's the issue

Reply:
If I drop the replacement attempt completely, and just set '3' lines in the yaml, this works fine, so I don't think there is a problem here.

Reply:
agreed, it actually looks like the replacement isn't happening at all in this case